File: src/main/java/team404/MaterialLoader.java
package team404;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.bukkit.Bukkit;
import team404.models.RequiredResource;

import java.io.File;
import java.io.IOException;
import java.util.List;

public class MaterialLoader {
    private static final ObjectMapper objectMapper = new ObjectMapper();

    public static void loadMaterials(File dataFolder) {
        File materialsFolder = new File(dataFolder, "materials");
        if (materialsFolder.mkdirs()) {
            Bukkit.getLogger().info("Folder %s was created".formatted(materialsFolder.getName()));
        }
        // Список файлов в папке materials
        File[] materialFiles = materialsFolder.listFiles();

        if (materialFiles != null) {
            for (File file : materialFiles) {
                if (file.isFile() && file.getName().endsWith(".json")) {
                    String materialListKey = file.getName().replace(".json", "");
                    // Загружаем список материалов из файла
                    List<RequiredResource> materialList = loadMaterialList(file);
                    // Добавляем список в хранилище
                    MaterialGenerator.addMaterialList(materialListKey, materialList);
                    Bukkit.getLogger().info("Loaded %s with %s elements"
                            .formatted(materialListKey, materialList.size()));
                }
            }
        }
        Bukkit.getLogger().info("Total materials lists loaded - %s"
                .formatted(MaterialGenerator.getMaterialLists().size()));
    }

    private static List<RequiredResource> loadMaterialList(File file) {
        try {
            if (!file.exists()) {
                return List.of();
            }
            return objectMapper.readValue(file, new TypeReference<>() {
            });
        } catch (IOException e) {
            Bukkit.getLogger().info("File %s was not loaded".formatted(file.getName()));
            return List.of();
        }
    }
}

File: src/main/kotlin/team404/service/MaterialLoader.kt
package team404.service

import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.ObjectMapper
import org.slf4j.LoggerFactory
import team404.model.RequiredResource
import java.io.File
import java.io.IOException

object MaterialLoader {
    private val objectMapper = ObjectMapper()

    private val logger = LoggerFactory.getLogger(MaterialLoader::class.java)

    fun loadMaterials(dataFolder: File) {
        val materialsFolder = File(dataFolder, "materials")
        if (materialsFolder.mkdirs()) {
            logger.info("Folder ${materialsFolder.name} was created")
        }
        // Список файлов в папке materials
        val materialFiles = materialsFolder.listFiles()

        materialFiles?.forEach { file ->
            if (file.isFile && file.name.endsWith(".json")) {
                val materialListKey = file.name.replace(".json", "")
                // Загружаем список материалов из файла
                val materialList = loadMaterialList(file)
                // Добавляем список в хранилище
                MaterialGenerator.addMaterialList(materialListKey, materialList)
                logger.info("Loaded $materialListKey with ${materialList.size} elements")
            }
        }
        logger.info("Total materials lists loaded - ${MaterialGenerator.materialLists.size}")
    }

    private fun loadMaterialList(file: File): List<RequiredResource> {
        return try {
            if (!file.exists()) {
                emptyList()
            } else {
                objectMapper.readValue(file, object : TypeReference<List<RequiredResource>>() {})
            }
        } catch (e: IOException) {
            logger.info("File ${file.name} was not loaded")
            emptyList()
        }
    }
}
